### PR TITLE
Add frequency_impulse_response

### PR DESCRIPTION
**Pull request by mthrok**
This commit adds `frequency_impulse_response` function, which generates filter from desired frequency response.

[Example](https://output.circle-artifacts.com/output/job/5233fda9-dadb-4710-9389-7e8ac20a062f/artifacts/0/docs/tutorials/filter_design_tutorial.html#frequency-sampling)

### CODE REVIEW

File: torchaudio/prototype/functional/_dsp.py [hunk 1]
```diff
@@ -286,3 +286,23 @@ def sinc_impulse_response(cutoff: torch.Tensor, window_size: int = 513, high_pas
...
+        Tensor: Impulse response. Shape `(..., 2 * (num_fft_bins - 1))`
+    """
+    if magnitudes.min() < 0.0:
+        # Negative magnitude does not make sense but allowing so that autograd works
+        # around 0.
+        # Should we raise error?
+        warnings.warn("The input frequency response should not contain negative values.")
+    ir = torch.fft.fftshift(torch.fft.irfft(magnitudes), dim=-1)
+    device, dtype = magnitudes.device, magnitudes.dtype
+    window = torch.hann_window(ir.size(-1), periodic=False, device=device, dtype=dtype).expand_as(ir)
```
Comment (hwangjeff) [inline torchaudio/prototype/functional/_dsp.py:307]: does it have to be a hann window specifically?

Comment (mthrok) [reply torchaudio/prototype/functional/_dsp.py:307]: No, this is no different than the standard approach of windowing. Other windows can be used. Hann window is the standard one.

Comment (mthrok) [reply torchaudio/prototype/functional/_dsp.py:307]: by standard approach, I mean smoothing the side effect of bound effects.

Comment (hwangjeff) [reply torchaudio/prototype/functional/_dsp.py:307]: got it. in that case, perhaps we can consider adding an argument similar to `window_fn` for [`Spectrogram`](https://pytorch.org/audio/main/generated/torchaudio.transforms.Spectrogram.html#torchaudio.transforms.Spectrogram) that allows for customizing the window

**Review by hwangjeff — APPROVED** [review]

**Pull request closed without merging**